core/utils/checksum.h

```diff
@@ -109,9 +110,11 @@ static inline uint32_t CalculateSum(const void *buf, size_t len) {
   // Reduce 64-bit unsigned integer to 32-bit unsigned integer
   // Carry may happens, but no need to complete reduce here
   sum64 = (sum64 >> 32) + (sum64 & 0xFFFFFFFF);
+#endif
 
   // Repeat 16-bit one's complement sum (at sum64)
-  const uint16_t *buf16 = reinterpret_cast<const uint16_t *>(buf64);
+  typedef uint16_t __attribute__((__may_alias__)) u16_p;
```

Comment:
I see. What I did was just build test (with `-m32` option to g++)i, as I don't have a 32-bit env to do runtime test. While it's entirely cosmetic, perhaps we can use `[[gnu:may_alias]]` instead?